File: app/src/main/java/myarena/api/API_запрос.java
package myarena.api;

/**
 * Базовый класс запроса к MyArena.ru API
 */
public class API_запрос {
    String query;
    String token;
    protected final String baseURL = "https://www.myarena.ru/api.php?";

    public API_запрос(String query, String token) {
        this.query = query;
        this.token = token;
    }

    /**
     * @return Возвращает GET-запрос, который нужно отправить на сервер MyArena.ru
     */
    public String toHTTPs() {
        return baseURL + "query=" + query + "&token=" +  token;
    }
}

File: app/src/main/java/myarena/api/API_запрос.kt
package myarena.api

/**
 * Базовый класс запроса к MyArena.ru API
 */
open class API_запрос(query: String, token: String) {
    internal var query = ""
    internal var token = ""

    init {
        this.query = query
        this.token = token
    }

    /**
     * @return Возвращает GET-запрос, который нужно отправить на сервер MyArena.ru
     */
    open fun toHTTPs(): String {
        return "${baseURL}query=$query&token=$token"
    }
}
